x-pack/plugin/sql/sql-proto/src/main/java/org/elasticsearch/xpack/sql/proto/Version.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.sql.proto;
+
+import java.security.InvalidParameterException;
+
+public class Version implements Comparable<Version>{
+
+    public final int id;
+    public final String version; // originally provided String representation
+    public final byte major;
+    public final byte minor;
+    public final byte revision;
+
+    public static final int REVISION_MULTIPLIER = 100;
+    public static final int MINOR_MULTIPLIER = REVISION_MULTIPLIER * REVISION_MULTIPLIER;
+    public static final int MAJOR_MULTIPLIER = REVISION_MULTIPLIER * MINOR_MULTIPLIER;
+
+
+
+    public static final Version V_7_7_0 = new Version(7, 7, 0);
```

Comment:
It could make sense, yes, but I imagine that we'll need to log a new version here with each "breaking" change we add in the protocol. If we'd instrument somehow these changes, referring to the versions by their number might be easier. But we could also refer to the versions by a name that conveys the gist of the breaking change, true. Wdyt?

Reply:
While I would prefer a name, I think using the versions gives us excellent flexibility in the long term. I'd opt towards adding a comment plus an utility method (that uses the field internally) with a verbose name:
```
public boolean supportsVersioning(Version version) {
  return version.compareTo(V_7_...) >= 0;
}
```